fix(dav): fix event birthday alarms not being updated

tcitworld opened the pull request:
In https://github.com/nextcloud/server/pull/33251 the default offset for a birthday event alarm was changed to 9AM on the day of the event.

However the `birthdayEvenChanged` method did not account for alarm changes, so it was never propagated to existing birthday events, which were kept on midnight.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

tcitworld committed:
fix(dav): fix event birthday alarms not being updated

In https://github.com/nextcloud/server/pull/33251 the default offset for
a birthday event alarm was changed to 9AM on the day of the event.

However the birthdayEvenChanged method did not account for alarm
changes, so it was never propagated to existing birthday events, which
were kept on midnight.

Signed-off-by: Thomas Citharel <tcit@tcit.fr>

tcitworld commented:
/backport to stable27

tcitworld commented:
/backport to stable26

tcitworld commented:
/backport to stable25

nfebe reviewed (approved):
LGTM

tcitworld commented:
Should we add a repair job just like `RegenerateBirthdayCalendars` to fix all existing events? Or just edit the app config value from `regeneratedBirthdayCalendarsForYearFix` to something like `regeneratedBirthdayCalendarsForAlarmFix` so that it's triggered again?

miaulalala commented:
> Should we add a repair job just like `RegenerateBirthdayCalendars` to fix all existing events? Or just edit the app config value from `regeneratedBirthdayCalendarsForYearFix` to something like `regeneratedBirthdayCalendarsForAlarmFix` so that it's triggered again?

Repair step sounds good to me.

tcitworld committed:
fix(dav): add a new config key to check to retrigger regenerating birthday calendars

So that birthday calendars are immediately updated and we don't need to wait for user to change a
card or disable/enable the calendar. We reuse the existing RegenerateBirthdayCalendars repair step
instead of adding a new one.

Signed-off-by: Thomas Citharel <tcit@tcit.fr>

tcitworld commented:
Rebased

tcitworld commented:
/backport to stable28

skjnldsv commented:
psalm says 
![No Way Snl GIF](https://github.com/nextcloud/server/assets/14975046/f9f4d210-7f61-4fee-bea2-f5b9856308b4)
